src/components/FileList.jsx

```diff
@@ -1,14 +1,81 @@
-import React from 'react'
+import React, { PureComponent } from 'react'
 import { translate } from '../lib/I18n'
 
-import File from '../components/File'
+import InfiniteLoader from 'react-virtualized/dist/commonjs/InfiniteLoader'
+import AutoSizer from 'react-virtualized/dist/commonjs/AutoSizer'
+import List from 'react-virtualized/dist/commonjs/List'
+import File, { FilePlaceholder } from '../components/File'
 
-const FileList = ({
-  t, f, displayedFolder, files, selected, selectionModeActive, onFileEdit, onFolderOpen, onFileOpen, onFileToggle, showActionMenu
-}) => (
-  <div>
-    {files.map((file, idx) => (
+const STATUS_LOADING = 1
+const STATUS_LOADED = 2
+
+const FETCH_LIMIT = 50
+const ROW_HEIGHT = 50
+
+class FileList extends PureComponent {
```

Comment:
`PureComponent` is used in react-virtualized samples, so I presumed it could have a performance impact.